src/shared/frame-rpc.js

```diff
@@ -117,45 +150,106 @@ export class RPC {
     if (this._destroyed) {
       return;
     }
+
     const seq = this._sequence++;
-    if (typeof args[args.length - 1] === 'function') {
-      this._callbacks[seq] = args[args.length - 1];
+    const finalArg = args[args.length - 1];
+    if (typeof finalArg === 'function') {
+      this._callbacks[seq] = finalArg;
       args = args.slice(0, -1);
     }
-    this.destFrame.postMessage(
-      {
-        protocol: 'frame-rpc',
-        version: VERSION,
-        sequence: seq,
-        method,
-        arguments: args,
-      },
-      this.origin
-    );
+
+    /** @type {RequestMessage} */
+    const message = {
+      arguments: args,
+      method,
+      protocol: PROTOCOL,
+      sequence: seq,
+      version: VERSION,
+    };
+
+    if (this._port) {
+      this._port.postMessage(message);
+    }
+
+    // Deprecated - Remove after MessagePort conversion
+    if (this.destFrame) {
+      this.destFrame.postMessage(message, this.origin);
+    }
   }
 
   /**
-   * @param {Message} msg
+   * Validate sender
+   *
+   * @param {MessageEvent} event
+   * @deprecated
    */
-  _handle(msg) {
-    if (this._destroyed) {
+  _isValidSender(event) {
+    if (event.source !== this.destFrame) {
+      return false;
+    }
+    if (this.origin !== '*' && event.origin !== this.origin) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Validate message
+   *
+   * @param {MessageEvent} event
+   * @return {null|Message}
+   */
+  _parseMessage({ data }) {
+    if (!data || typeof data !== 'object') {
+      return null;
+    }
+    if (data.protocol !== PROTOCOL) {
```

Comment:
All these statements were combined into one. I can combined back. I left them like this because it is more clear for developers and easier to check code coverage.

Reply:
> easier to check code coverage.

That's true. OK, let's leave as is.